Dc strategy

dcowden opened the pull request:
Adds strategies to separate selection of auto paths from the other code.
adds code to select strategies from buttons, switches, or an integer from the smart dashboard or robot preferences.

Plaba commented on Macarena2018/src/main/java/frc/team281/robot/AutoModeSelectorWidget.java:
```diff
@@ -0,0 +1,119 @@
+	
+	public static final String AUTO_SELECTOR_WIDGET = "AutoModeSelectorWidget";
+	
+	private double leftSwitchPriority = 1;
```
Shouldn't they be ints?

dcowden commented:
@mandrews281 Fairly happy with how this turned out.  Highlights:
   Used the four pictures you had, and called them 'FieldPoses'. Assumed we'd have a boolean for each picture, and then computed the result in all the cases

Added the code to build command differently based on targeting switch.
Added a bunch of convenience methods to make it easer to do the logic, but it worked well i think.
